addons/mrp/models/mrp_production.py

```diff
@@ -574,7 +574,10 @@ def onchange_product_id(self):
     def _onchange_product_qty(self):
         for workorder in self.workorder_ids:
             workorder.product_uom_id = self.product_uom_id
-            workorder.duration_expected = workorder._get_duration_expected()
+            if self.bom_id:
+                workorder._onchange_expected_duration()
```

Comment:
I would rather modify https://github.com/odoo/odoo/blob/master/addons/mrp/models/mrp_workorder.py#L687
to use the duration_expected_working as if the alternate workcenter was set